daprdocs/content/en/dotnet-sdk-docs/dotnet-actors/dotnet-actors-usage.md

```diff
@@ -150,6 +153,12 @@ public void ConfigureServices(IServiceCollection services)
     services.AddSingleton<BankService>();
 }
 ```
+> [!IMPORTANT]  
+> When registering actors, note the return type requirements for the methods inside the interfaces:
+> * Pattern 1: `options.Actors.RegisterActor<MyActor>()`
+>     * In this case, all interfaces implemented by `MyActor` must have methods that return only `Task` or `Task<T>`. This applies to every method in all interfaces `MyActor` implements.
+> * Pattern 2: `options.Actors.RegisterActor<IMyActor, MyActor>()`
+>     * Here, only the `IMyActor` interface is required to have methods that return `Task` or `Task<T>`. Other interfaces `MyActor` are not subject to this restriction.
```

Comment:
If we force the use of `IActor` interface we can add a Diagnostics to block the compilation when the derived class contains methods that doesn't return `Task<T>`. Similar to what is done in the actors' source generator for the `CancellationToken` ([example](https://github.com/dapr/dotnet-sdk/blob/a50e6de3063cbc85649b56aa357d705888b96489/src/Dapr.Actors.Generators/Diagnostics/CancellationTokensMustBeTheLastArgument.cs))

I usually don't like to find out particularities from documentation, if possible I prefer my editor to guide me.

What do you think about it?